### DISCUSSION TITLE
All Comments/Forum posts + more

### DISCUSSION THREAD

**Discussion by zappgame** [ScratchAddons/ScratchAddons#5266] [Ideas] [answered]
Does S.A. have access to every Comment a person has made? If so, They should add an option to see every comment or forum post a person has made to see if a person is a troll/is innapropriate/mean. There should also be an option to see if somebody is online or not and maybe a thing that says "Working on a project" above the avatar if somebody is inside the editor, and Maybe adding thumbnails/screenshots for W.I.P. projects next to the "What I'm working on" section of a profile. and also a thing where if you hover with your mouse above the heart/star/view/remix count, you see who did that action.

**Comment by CST1229** [comment, marked as answer]
> Does S.A. have access to every Comment a person has made?

No. Might be somewhat possible if someone creates a comment database (like how [ScratchDB](https://scratchdb.lefty.one) indexes forum posts), but it will be slow to index and will probably be incomplete. You can see a user's forum posts with a number of tools, like [ocular](https://ocular.jeffalo.net) ([example](https://ocular.jeffalo.net/users/griffpatch)).  
The Scratch Team might have access to all of a user's comments, but I don't think that info would ever become available to all users.

> There should also be an option to see if somebody is online or not and maybe a thing that says "Working on a project" above the avatar if somebody is inside the editor

Online indicators for people also using the extension are being discussed (on how to store the info), I don't know about rich presence. It is impossible to see if someone is online if they don't use the extension (to send their online indicator).

> Maybe adding thumbnails/screenshots for W.I.P. projects next to the "What I'm working on" section of a profile

Maybe, but we'd need a way to store the images. You could just link the project thumbnails in your WIWO (you can see unshared thumbnails currently); just note that [people can also currently access unshared projects if they know the ID](https://docs.turbowarp.org/unshared-projects), so you might want to put the images in a separate project containing only the image.

> also a thing where if you hover with your mouse above the heart/star/view/remix count, you see who did that action

Impossible, the Scratch API doesn't provide that info.
